### PR TITLE
Refactor printObject

### PR DESCRIPTION
**Pull request by thorn0**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/print/object.js [hunk 1]
```diff
@@ -80,11 +90,11 @@ function printObject(path, options, print) {
             property.value.type === "ArrayPattern")
       )) ||
     (node.type !== "ObjectPattern" &&
-      firstProperty &&
+      propsAndLoc.length > 0 &&
       hasNewlineInRange(
         options.originalText,
         locStart(node),
-        locStart(firstProperty)
+        locStart(propsAndLoc[0].node)
```
Comment (fisker) [inline src/language-js/print/object.js:97]:
```suggestion
        propsAndLoc[0].loc
```

**Review by fisker — APPROVED** [review]
LGTM. Except I don't like nest ternary..

**Comment by fisker** [comment]
https://github.com/prettier/prettier/blob/d545e5ffda6a19e2e6a014c2761043fcea367dfb/src/language-js/print/object.js#L204-L210 Doesn't look right to me, number = 0 doesn't mean it's first argument.

**Comment by thorn0** [comment]
I agree. It looks suspicious.

**Pull request merged**